src/Symfony/Component/Translation/FallbackLocaleProviderInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Component\Translation\Exception\InvalidArgumentException;
+
+/**
+ * For a given locale, implementations provide the list of alternative locales to
+ * try when a translation cannot be found.
+ *
+ * @author Matthias Pigulla <mp@webfactory.de>
+ */
+interface FallbackLocaleProviderInterface
+{
+    /**
+     * @internal
+     *
+     * @return string[]
+     */
+    public function getUltimateFallbackLocales(): array;
```

Comment:
Given that it is `@internal`, should it even be part of the interface?

Reply:
Do you have callees from external? If not, feel free to indeed remove from the interface. If we have calls from outside the class, do we really need those calls because calling internal classes might be considered a violation.